File: apps/android/core/testing/src/main/kotlin/app/gauja/core/testing/FixtureLoader.kt
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
package app.gauja.core.testing

import java.io.File

object FixtureLoader {
    fun artwork(): ByteArray = root().resolve("design/assets/test/poster.png").readBytes()

    private fun root(): File =
        generateSequence(File(requireNotNull(System.getProperty("user.dir")))) { it.parentFile }
            .first { it.resolve("api/UPSTREAM_COMMIT").isFile }

    fun read(relative: String): String {
        require(!relative.contains("..") && !relative.startsWith("/"))
        val root =
            generateSequence(File(requireNotNull(System.getProperty("user.dir")))) { it.parentFile }
                .first { it.resolve("api/UPSTREAM_COMMIT").isFile }
        return root.resolve("api/fixtures/3.4.1/$relative").readText()
    }
}

File: apps/ios/Packages/Testing/Sources/GaujaTesting/FixtureLoader.swift
// SPDX-FileCopyrightText: 2026 Gauja contributors
// SPDX-License-Identifier: AGPL-3.0-or-later
import Common
import Foundation

public enum FixtureLoader {
    public static func read(_ relative: String) throws -> Data {
        guard !relative.contains(".."), !relative.hasPrefix("/") else { throw AppError.validation }
        var directory = URL(fileURLWithPath: #filePath).deletingLastPathComponent()
        while !FileManager.default.fileExists(atPath: directory.appendingPathComponent("api/UPSTREAM_COMMIT").path) {
            guard directory.path != "/" else { throw AppError.notFound }
            directory.deleteLastPathComponent()
        }
        return try Data(contentsOf: directory.appendingPathComponent("api/fixtures/3.4.1/" + relative))
    }
}
